libusb/os/linux_usbfs.c

```diff
@@ -1463,6 +1463,13 @@ static int op_open(struct libusb_device_handle *handle)
 	return usbi_add_pollfd(HANDLE_CTX(handle), hpriv->fd, POLLOUT);
 }
 
+static int op_open2(struct libusb_device_handle *handle, int fd)
+{
+	struct linux_device_handle_priv *hpriv = _device_handle_priv(handle);
+	hpriv->fd = fd;
+	return usbi_add_pollfd(HANDLE_CTX(handle), hpriv->fd, POLLOUT);
+}
+
```

Comment:
This gives a `warning: 'op_open2' defined but not used [-Wunused-function]` on Linux, so maybe you'd want to `#ifdef __ANDROID__` this part too.